airbyte-server/src/main/java/io/airbyte/server/apis/ConnectionApiImpl.java

```diff
@@ -0,0 +1,189 @@
+import io.airbyte.api.model.generated.ConnectionCreate;
+import io.airbyte.api.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.model.generated.ConnectionRead;
+import io.airbyte.api.model.generated.ConnectionReadList;
+import io.airbyte.api.model.generated.ConnectionSearch;
+import io.airbyte.api.model.generated.ConnectionState;
+import io.airbyte.api.model.generated.ConnectionStateCreateOrUpdate;
+import io.airbyte.api.model.generated.ConnectionStateType;
+import io.airbyte.api.model.generated.ConnectionUpdate;
+import io.airbyte.api.model.generated.JobInfoRead;
+import io.airbyte.api.model.generated.WorkspaceIdRequestBody;
+import io.airbyte.commons.version.AirbyteVersion;
+import io.airbyte.config.Configs.WorkerEnvironment;
+import io.airbyte.config.helpers.LogConfigs;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.SecretsRepositoryReader;
+import io.airbyte.config.persistence.SecretsRepositoryWriter;
+import io.airbyte.config.persistence.StatePersistence;
+import io.airbyte.persistence.job.JobPersistence;
+import io.airbyte.persistence.job.WorkspaceHelper;
+import io.airbyte.server.handlers.ConnectionsHandler;
+import io.airbyte.server.handlers.DestinationDefinitionsHandler;
+import io.airbyte.server.handlers.DestinationHandler;
+import io.airbyte.server.handlers.JobHistoryHandler;
+import io.airbyte.server.handlers.OperationsHandler;
+import io.airbyte.server.handlers.SchedulerHandler;
+import io.airbyte.server.handlers.SourceDefinitionsHandler;
+import io.airbyte.server.handlers.SourceHandler;
+import io.airbyte.server.handlers.StateHandler;
+import io.airbyte.server.handlers.WebBackendConnectionsHandler;
+import io.airbyte.server.scheduler.EventRunner;
+import io.airbyte.server.scheduler.SynchronousSchedulerClient;
+import io.airbyte.validation.json.JsonSchemaValidator;
+
+public class ConnectionApiImpl implements ConnectionApi {
+
+  private final ConnectionsHandler connectionsHandler;
+  private final OperationsHandler operationsHandler;
+  private final SchedulerHandler schedulerHandler;
+  private final StateHandler stateHandler;
+  private final WebBackendConnectionsHandler webBackendConnectionsHandler;
+
+  public ConnectionApiImpl(final ConfigRepository configRepository,
+                           final JobPersistence jobPersistence,
+                           final TrackingClient trackingClient,
+                           final EventRunner eventRunner,
+                           final SecretsRepositoryReader secretsRepositoryReader,
+                           final SecretsRepositoryWriter secretsRepositoryWriter,
+                           final SynchronousSchedulerClient synchronousSchedulerClient,
+                           final WorkerEnvironment workerEnvironment,
+                           final LogConfigs logConfigs,
+                           final StatePersistence statePersistence,
+                           final AirbyteVersion airbyteVersion) {
```

Comment:
As this class really only cares about
```
  private final ConnectionsHandler connectionsHandler;
  private final OperationsHandler operationsHandler;
  private final SchedulerHandler schedulerHandler;
  private final StateHandler stateHandler;
  private final WebBackendConnectionsHandler webBackendConnectionsHandler;
```
Should we expect these^ fields in the constructor instead of the objects used to generate these fields?

Reply:
I'm following the existing pattern here, happy to change it. I wanted to limit the creation of object in the `ServerFactory.create` method (I just realized that I forget to set the value there). That being said, I agree with you it makes this class hard to test (maybe that's why there were no test for the configurationApi)